hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/MockAbfsInputStream.java

```diff
@@ -125,9 +147,96 @@ public void induceConnectionException() {
     mockConnectionException = true;
   }
 
+  public void disableAlwaysOnFastpathTestMock() {
+    disableForceFastpathMock = true;
+    ((MockAbfsClient)client).forceFastpathReadAlways = false;
+  }
+
   public void resetAllMockErrStates() {
     errStatus = 0;
     mockRequestException = false;
     mockConnectionException = false;
   }
+
+  public static AbfsFastpathSession getStubAbfsFastpathSession(final AbfsClient client,
+      final String path,
+      final String eTag,
+      TracingContext tracingContext,
+  AbfsFastpathSessionInfo ssnInfo) throws Exception {
+    AbfsFastpathSession mockSession = getStubAbfsFastpathSession(client, path, eTag, tracingContext);
+    // set the sessionInfo so that fileHandle and connectionMode are set
+    // (session token and expiry will also get set but they will be rewritten)
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "fastpathSessionInfo", ssnInfo);
+    // Overwrite session token and expiry so that refresh of the token is
+    // triggered as well.
+    mockSession.updateAbfsFastpathSessionToken(ssnInfo.getSessionToken(), ssnInfo.getSessionTokenExpiry());
+    return mockSession;
+  }
+
+  public static AbfsFastpathSession getStubAbfsFastpathSession(final AbfsClient client,
+      final String path,
+      final String eTag,
+      TracingContext tracingContext) throws Exception {
+
+    AbfsFastpathSession mockSession = mock(AbfsFastpathSession.class);
+    Logger log = LoggerFactory.getLogger(AbfsInputStream.class);
+    double session_refresh_internal_factor = 0.75;
+    ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(1);
+    ReentrantReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+    // override fields
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "LOG", log);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "SESSION_REFRESH_INTERVAL_FACTOR", session_refresh_internal_factor);
+    mockSession = TestMockHelpers.setClassField(AbfsFastpathSession.class,
+        mockSession, "client", client);
```

Comment:
nit: using constants instead of strings here?

Reply:
This is a wont fix. This is test mock code and there is no value add in adding a lot of static variables for this.